application-numbered-content-figures/application-numbered-content-figures-ui/src/main/resources/NumberedFigures/Code/NumberedFigures.xml

```diff
@@ -491,15 +486,20 @@ ${H}xwikicontent div[data-cke-display-name="${figureMacroDisplayName}"] figure:n
 #set($isNumberedFiguresEnabled = $services.numberedfigures.isNumberedFiguresEnabled())
 #if ($isNumberedFiguresEnabled)
   #set ($origdoc = $xwiki.getDocument('NumberedFigures.Code.NumberedFigures'))
+  #set ($commondoc = $xwiki.getDocument('NumberedCommon.Code.NumberedCommon'))
   #set ($locale = $services.localization.getCurrentLocale())
   #set($ssxHref = $origdoc.getURL('ssx', $escapetool.url({
     'isNumberedFiguresEnabled': $isNumberedFiguresEnabled,
     'locale': $locale,
     'hash': $services.numberedfigures.getDisplayDataHash($locale),
     'docVersion': $origdoc.getVersion()
   })))
+  #set($ssxHrefCommon = $commondoc.getURL('ssx', $escapetool.url({
+    'hash': $services.numberedcommon.hash($locale)
+  })))
   {{html clean='false'}}
   &lt;link rel='stylesheet' type='text/css' href="$escapetool.xml($ssxHref)"/&gt;
+  &lt;link rel='stylesheet' type='text/css' href="$escapetool.xml($ssxHrefCommon)"/&gt;
```

Comment:
I would have at the very least expected this to be wrapped in some script service, and ideally also not duplicated with numbered figures (maybe the script service could store some request property if the SSX has been added already?).

Reply:
The script service is a good point.
Note: if the same `<link rel="stylesheet" type="text/css" href="/xwiki/bin/ssx/NumberedCommon/Code/NumberedCommon?hash=...">` is added twice in the code, it leads to a single http request to fetch the CSS.
I'm not sure if we need to make the code more complex for this.

Reply:
Okay, then let's maybe not overcomplicate this, still it would be nice to avoid a bit the code duplication.

Reply:
Actually, a correct use of `org.xwiki.skinx.SkinExtension#use(java.lang.String, java.util.Map<java.lang.String,java.lang.Object>)` does the job. The only risk is to have two different hashs, leading to two imports in the following case:
For a single page load
1. call `SkinExtension#use` with hash H1
2. during the page load, somebody install an extension providing new keys, leading to a new hash H2
2. a second call to `SkinExtension#use` is done, with hash H2

But this is very much a cornercase.